cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -97,8 +97,18 @@ function(arrow_create_merged_static_lib output_target)
   endforeach()
 
   if(APPLE)
-    set(BUNDLE_COMMAND "libtool" "-no_warning_for_no_symbols" "-static" "-o"
-                       ${output_lib_path} ${all_library_paths})
+    execute_process(COMMAND "libtool" -V
+    OUTPUT_VARIABLE LIBTOOL_V_OUTPUT
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if("${LIBTOOL_V_OUTPUT}" MATCHES ".*cctools-([0-9.]+).*")
+      # this is the macOS provided libtool
+      set(BUNDLE_COMMAND "libtool" "-no_warning_for_no_symbols" "-static" "-o"
+      ${output_lib_path} ${all_library_paths})
+    else()
+      # this is the GNU libtool, but we should be able to find the macos libtool we want
+      set(BUNDLE_COMMAND "/usr/bin/libtool" "-no_warning_for_no_symbols" "-static" "-o"
+          ${output_lib_path} ${all_library_paths})
```

Comment:
This could probably be factored into more idiomatic cmake with `find_program()` or the like.

I also should make the if/else only identify the path and then keep the options on one line too.